src/status_im/ui2/screens/chat/components/chat_item.cljs

```diff
@@ -0,0 +1,199 @@
+(ns status-im.ui2.screens.chat.components.chat-item
+  (:require [clojure.string :as string]
+            [re-frame.core :as re-frame]
+            [status-im.constants :as constants]
+            [status-im.i18n.i18n :as i18n]
+            [status-im.ui.components.badge :as badge]
+            [quo.design-system.colors :as colors]
+            [status-im.ui.components.chat-icon.screen :as chat-icon.screen]
+            [status-im.ui.screens.home.styles :as styles]
+            [status-im.ui.components.icons.icons :as icons]
+            [status-im.utils.core :as utils]
+            [status-im.utils.datetime :as time]
+            [status-im.ui.components.chat-icon.styles :as chat-icon.styles]
+            [quo2.components.markdown.text :as quo2.text]
+            [quo2.foundations.colors :as quo2.colors]
+            [quo.react-native :as rn]))
+
+(defn preview-label [label-key label-fn]
+  [rn/text {:style               styles/last-message-text
+            :accessibility-label :no-messages-text
+            :number-of-lines     1}
+   (i18n/label label-key label-fn)])
+
+(def max-subheader-length 100)
+
+(defn truncate-literal [literal]
+  (when literal
+    (let [size (min max-subheader-length (.-length literal))]
+      {:components (.substring literal 0 size)
+       :length     size})))
+
+(defn add-parsed-to-subheader [acc {:keys [type destination literal children]}]
+  (let [result (case type
+                 "paragraph"
+                 (reduce
+                  (fn [{:keys [_ length] :as acc-paragraph} parsed-child]
+                    (if (>= length max-subheader-length)
+                      (reduced acc-paragraph)
+                      (add-parsed-to-subheader acc-paragraph parsed-child)))
+                  {:components [rn/text]
+                   :length     0}
+                  children)
+
+                 "mention"
+                 {:components [rn/text @(re-frame/subscribe [:contacts/contact-name-by-identity literal])]
+                  :length     4} ;; we can't predict name length so take the smallest possible
+
+                 "status-tag"
+                 (truncate-literal (str "#" literal))
+
+                 "link"
+                 (truncate-literal destination)
+
+                 (truncate-literal literal))]
+    {:components (conj (:components acc) (:components result))
+     :length     (+ (:length acc) (:length result))}))
+
+(defn render-subheader
+  "Render the preview of a last message to a maximum of max-subheader-length characters"
```

Comment:
In docstrings, it's recommended to use backticks to surround vars. Tools often understand this and use different colors and even allow for automatic jump to source.

```clojure
"Render the preview of a last message to a maximum of `max-subheader-length` characters"
```